src/QuadForm/Torsion.jl

```diff
@@ -779,6 +817,134 @@ function has_complement(i::TorQuadModuleMor)
+#
+#  Arithmetic of maps
+#
+################################################################################
+
+@doc raw"""
+    +(f::TorQuadModuleMor, g::TorQuadModuleMor) -> TorQuadModuleMor
+
+Given two abelian group homomorphisms `f` and `g` between the same torsion
+quadratic modules `T` and `U`, return the pointwise sum `h` of `f` and `g`
+which sends every elements `a` of `T` to $h(a) := f(a) + g(a)$.
+"""
+function Base.:(+)(f::TorQuadModuleMor, g::TorQuadModuleMor)
+  @req domain(f) === domain(g) "f and g must have the same domain"
+  @req codomain(f) === codomain(g) "f and g must have the same codomain"
+  return hom(domain(f), codomain(f), TorQuadModuleElem[f(a)+g(a) for a in gens(domain(f))])
+end
+
+@doc raw"""
+    -(f::TorQuadModuleMor) -> TorQuadModuleMor
+
+Given an abelian group homomorphism `f` between two torsion quadratic modules
+`T` and `U`, return the pointwise opposite morphism `h` of `f` which sends every
+elements `a` of `T` to $h(a) := -f(a)$.
+"""
+function Base.:(-)(f::TorQuadModuleMor)
+  return hom(domain(f), codomain(f), TorQuadModuleElem[-f(a) for a in gens(domain(f))])
+end
+
+@doc raw"""
+    -(f::TorQuadModuleMor, g::TorQuadModuleMor) -> TorQuadModuleMor
+
+Given two abelian group homomorphisms `f` and `g` between the same torsion
+quadratic modules `T` and `U`, return the pointwise difference `h` of `f` and
+`g` which sends every elements `a` of `T` to $h(a) := f(a) - g(a)$.
+"""
+function Base.:(-)(f::TorQuadModuleMor, g::TorQuadModuleMor)
+  @req domain(f) === domain(g) "f and g must have the same domain"
+  @req codomain(f) === codomain(g) "f and g must have the same codomain"
+  return hom(domain(f), codomain(f), TorQuadModuleElem[f(a)-g(a) for a in gens(domain(f))])
+end
+
+@doc raw"""
+    *(a::IntegerUnion, f::TorQuadModuleMor) -> TorQuadModuleMor
+    *(f::TorQuadModuleMor, a::IntegerUnion) -> TorQuadModuleMor
+
+Given an abelian group homomorphism `f` between two torsion quadratic modules
+`T` and `U`, return the pointwise $a$-twist morphism `h` of `f` which sends every
+elements `b` of `T` to $h(b) := a*f(b)$.
+"""
+function Base.:(*)(a::IntegerUnion, f::TorQuadModuleMor)
+  return hom(domain(f), codomain(f), TorQuadModuleElem[a*f(b) for b in gens(domain(f))])
+end
+
+Base.:(*)(f::TorQuadModuleMor, a::IntegerUnion) = a*f
+
+@doc raw"""
+    ^(f::TorQuadModuleMor, n::Integer) -> TorQuadModuleMor
+
+Given an abelian group endomorphism `f` of a torsion quadratic module `T`
+return the $n$-fold self-composition of `f`.
+
+Note that `n` must be non-negative and $f^0$ is by default the identity map
+of the domain of `f` (see [`identity_map`](@ref)).
+"""
+function Base.:^(f::TorQuadModuleMor, n::Integer)
+  @req n >= 0 "n must be a positive integer"
+  @assert domain(f) === codomain(f) "f must be a self-map"
+  if n == 0
+    return id_hom(domain(f))
+  elseif n == 1
+    return f
+  else
+    k = 1
+    f2 = f
+    while k != n
+      f2 = compose(f2, f)
+      k += 1
+    end
+    return f2
+  end
```

Comment:
This is too inefficient. Just take a power of the underlying matrix.

Reply:
Indeed looks better. I modified the other functions using this idea, and by adding the missing similar functions for `GrpAbFinGenMap`: taking just the power is not enough, it should be reduced then w.r.t. to an snf/hnf of the codomain.